pkg/egressgateway/policy.go

```diff
@@ -177,20 +177,34 @@ func (gwc *gatewayConfig) deriveFromPolicyGatewayConfig(gc *policyGatewayConfig)
 
 // destinationMinusExcludedCIDRs will return, for a given policy, a list of all
 // destination CIDRs to which the excluded CIDRs have been subtracted.
-func (config *PolicyConfig) destinationMinusExcludedCIDRs() []*net.IPNet {
+func (config *PolicyConfig) destinationMinusExcludedCIDRs() []netip.Prefix {
 	if len(config.excludedCIDRs) == 0 {
 		return config.dstCIDRs
 	}
 
-	cidrs := []*net.IPNet{}
+	cidrs := []netip.Prefix{}
 
 	for _, dstCIDR := range config.dstCIDRs {
-		dstCIDRMinusExcludedCIDRs := []*net.IPNet{dstCIDR}
+		dstCIDRMinusExcludedCIDRs := []netip.Prefix{dstCIDR}
 		for _, excludedCIDR := range config.excludedCIDRs {
-			newDstCIDRMinuxExcludedCIDRs := []*net.IPNet{}
+			newDstCIDRMinuxExcludedCIDRs := []netip.Prefix{}
 			for _, cidr := range dstCIDRMinusExcludedCIDRs {
-				r, _, l := ip.PartitionCIDR(*cidr, *excludedCIDR)
-				newDstCIDRMinuxExcludedCIDRs = append(newDstCIDRMinuxExcludedCIDRs, append(r, l...)...)
+				rightNets, _, leftNets := ip.PartitionCIDR(*netipx.PrefixIPNet(cidr), *netipx.PrefixIPNet(excludedCIDR))
```

Comment:
Seems like a lot of unnecessary conversions here if we had a `PartitionCIDR` function that natively worked with `netip` types. Could you take a crack at implementing one?

Reply:
@christarazi I took a quick look at trying to replicate `PartitionCIDR()` for the netip pkg. It seems to be a fair bit of work. Since this PR is already quite large, are you OK if I create an issue to track this need and work on it separately?

Reply:
To be honest in that case, I don't really see the benefit of merging this PR in its current form because it will just introduce unnecessary churn into the tree by having a bunch of calls to convert `net` types back and forth, which will eat away at CPU cycles. What seems like we are uncovering here is that `pkg/egressgateway` has a lot more interdependence on `net.*` types so, maybe it makes more sense to revisit it when more of the surrounding code has been converted over to `netip`. Does that sound reasonable.

Reply:
Yes, I think that sounds reasonable. Should I close this PR for the time being?

Reply:
You can convert it into draft mode also. Up to you.